pyflow/graphics/window.py

```diff
@@ -546,3 +580,47 @@ def onMoveToItems(self):
     def setTheme(self, theme_index):
         """Set the theme of the application."""
         theme_manager().selected_theme_index = theme_index
+
+    def start_kernel(self):
+        """Start the kernel."""
+        current_window = self.activeMdiChild()
+        if current_window is not None:
+            current_window.scene.kernel.start()
+            self.statusbar.showMessage("Kernel started")
+        else:
+            self.statusbar.showMessage("No active window")
+
+    def interrupt_kernel(self):
+        """Interrupt the kernel."""
+        current_window = self.activeMdiChild()
+        if current_window is not None:
+            current_window.scene.kernel.interrupt()
+            self.statusbar.showMessage("Kernel interrupted")
+        else:
+            self.statusbar.showMessage("No active window")
+
+    def stop_kernel(self):
+        """Stop the kernel."""
+        current_window = self.activeMdiChild()
+        if current_window is not None:
+            current_window.scene.kernel.stop()
+            self.reset_block_states()
+            self.statusbar.showMessage("Kernel stopped")
+        else:
+            self.statusbar.showMessage("No active window")
+
+    def restart_kernel(self):
+        """Restart the kernel."""
+        current_window = self.activeMdiChild()
+        if current_window is not None:
+            current_window.scene.kernel.restart()
+            self.reset_block_states()
+            self.statusbar.showMessage("Kernel restarted")
+        else:
+            self.statusbar.showMessage("No active window")
```

Comment:
Those functions are nearly the same, try to reduce duplication by using an other (private) method